Update braintree-web and add paypal-checkout-components

jbuckner opened the pull request:
This PR does a few things things
1. Adds `paypal-checkout-components` declarations. This is a dependency for the Braintree PayPal provider.

2. Adds 3 payment providers for `braintree-web`: 
 - `venmo` 
 - `google-payment` 
 - `paypal-checkout`

3. Organizes the `braintree-web` interfaces into a directory. The entire `braintree-web` framework is quite large so adding more interfaces to `index.d.ts` was getting unwieldy. This splits up the modules into individual files and references them. No changes to the declarations were made and no tests had to change, just changing the structure.

4. Run `prettier` on the declarations to match the style guide.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [] Create it with `dts-gen --dt`, not by basing it on an existing project. <-- I tried doing this, but kept getting `Couldn't load module "@paypal/checkout-components".`. This is also for an older version of the `paypal-checkout-components` supported by Braintree
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/braintree/braintree-web/blob/master/src/index.js>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

danger-public commented:
<!--
  0 failure: 
  0 warning: 
  
  1 markdown notices
  DangerID: danger-id-Danger;
-->



Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## braintree-web ([<kbd>unpkg</kbd>](https://unpkg.com/browse/braintree-web@latest/))
was missing the following properties:
1. localPayment
1. masterpass
1. paymentRequest
1. vaultManager
1. visaCheckout

<details>
<summary>as well as these 1 other properties...</summary>
<p>preferredPaymentMethods</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7bbe00b5aad1dc90259f74d7116dbec4fb853bf1
</p>

jbuckner commented:
Closing this for some fixes

elibarzilay commented:
@CHLELA: re-ping
@jbuckner: [tslint tweaks](https://github.com/DefinitelyTyped/DefinitelyTyped/#linter)

jbuckner commented:
@elibarzilay I removed all but two tslint tweaks and cleaned up all of the resulting errors. I couldn't figure out how to make the `paypal` namespace work using just module exports so had to still use this format:
```
declare namespace paypal {}

export = paypal;
```

elibarzilay commented:
@jbuckner: I just checked, and you only have comment lines that are long; as for the namespace, did you see [the page it refers to](https://github.com/Microsoft/dtslint/blob/master/docs/export-just-namespace.md)?

jbuckner commented:
@elibarzilay Yeah I totally misread that error message 🤦 I thought it was saying a file length error because the file was so long.

I was able to figure out what I was doing wrong with the namespacing. I misunderstood how it worked. I updated it so I'm just doing regular exports and imports now without the `declare namespace paypal` and it's much cleaner. 😅

jbuckner commented:
@elibarzilay @CHLELA I updated the `braintree-web` declarations to use import/export module-style instead of `declare namespace`... it's much cleaner now! I also removed all of the `tslint` overrides that were in it and cleaned it up.

Mainly the only things I had to change in the tests were `var` to `const`, `function` to `=>`, and some line-length issues.

jamesg1 commented:
@jbuckner Thanks for this PR, nice to see paypal-checkout included now. We had `declare var paypal: any;` before in our types.

molayodecker commented on types/paypal-checkout-components/test/web.ts:
```diff
@@ -0,0 +1,36 @@
+import * as paypal from "paypal-checkout-components";
```
Does this test really work because I've been having issues since this was merged

jbuckner replied:
What issues are you having?
